### PR TITLE
Add support for payment sources

### PR DESCRIPTION
**Pull request by adam-invoiced**
Opening PR because of new unit tests and overloading the delete function for PaymentSource.

### CODE REVIEW

File: invoiced/objects.py [hunk 1]
```diff
@@ -329,9 +329,9 @@ class PaymentSource(CreateableObject, DeleteableObject, ListableObject):
     
     def delete(self):
         if self.object == "card":
-            self._endpoint = '/card/' + str(self.id)
+            self._endpoint = '/cards/' + str(self.id)
         elif self.object == "bank_account":
-            self._endpoint = '/bank_account/' + str(self.id)
+            self._endpoint = '/bank_accounts' + str(self.id)
```
Comment (jaredtking) [inline invoiced/objects.py]: I think you left out a slash here

**Comment by adam-invoiced** [comment]
Can potentially close this PR as a result of conversation yesterday, because this change will be totally redone.

**Comment by jaredtking** [comment]
I'd leave it open if you are going to commit changes on top of this branch. If not then you should close and open with a new branch.

**Comment by adam-invoiced** [comment]
Ok, new version implements Card and BankAccount classes with create() and delete() functions, a well as a generic PaymentSource class with list() function that returns a list of Card and BankAccount objects.

**Comment by jaredtking** [comment]
Is this ready to merge?

**Comment by adam-invoiced** [comment]
Yep!

**Comment by jaredtking** [comment]
Should we cut a new version or are there other changes in the queue?

**Comment by adam-invoiced** [comment]
No other pending changes; a new version would be appropriate.

**Pull request merged**